### PR TITLE
TW-1161: copy/paste with shortcuts in firefox

### PR DESCRIPTION
**Pull request by sherlockvn**
### Migration documents:
- [WillPopScope](https://docs.flutter.dev/release/breaking-changes/android-predictive-back#migrating-a-back-confirmation-dialog:%7E:text=route%20transitions.%0A%7D-,Migrating%20a%20back%20confirmation%20dialog,-WillPopScope%20was%20sometimes)
- [flutter_typeahead](https://pub.dev/packages/flutter_typeahead/changelog)
- [super_clipboard](https://pub.dev/packages/super_clipboard/changelog)

### Demo:

https://github.com/linagora/twake-on-matrix/assets/43041967/97d781ad-595f-4c8c-ae1a-703d535a8be2

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -215,6 +216,12 @@ msix_config:
   install_certificate: false
 
 dependency_overrides:
+  # remove when super_clipboard release a new version higher than 0.8.0
```
Comment (hoangdat) [inline pubspec.yaml:219]: add `todo` to have warning to change it

Comment (sherlockvn) [reply pubspec.yaml]: done

File: lib/pages/chat/input_bar/input_bar.dart [hunk 2]
```diff
@@ -334,109 +335,63 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+          textInputAction: textInputAction,
+          autofocus: autofocus,
+          style: InputBarStyle.getTypeAheadTextStyle(context),
+          controller: controller,
+          decoration: decoration,
+          focusNode: focusNode,
+          onChanged: (text) {
+            // fix for the library for now
+            // it sets the types for the callback incorrectly
+            onChanged!(text);
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:361]: WTF why can't check `onChanged` != null ?

File: lib/pages/chat/input_bar/input_bar.dart [hunk 3]
```diff
@@ -334,109 +335,63 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+          onTap: () async {
+            await Future.delayed(const Duration(milliseconds: 100));
+            FocusScope.of(context).requestFocus(focusNode);
+          },
+          onSubmitted: PlatformInfos.isMobile
+              ? (text) {
+                  // fix for library for now
+                  // it sets the types for the callback incorrectly
+                  onSubmitted!(text);
+                }
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:372]: Check `onSubmitted` != null

File: pubspec.yaml [hunk 4]
```diff
@@ -44,10 +44,7 @@ dependencies:
   flutter_secure_storage: ^7.0.1
   flutter_svg: ^0.22.0
   # FIXME: change to upstream when https://github.com/AbdulRahmanAlHamali/flutter_typeahead/pull/528 is merge
-  flutter_typeahead:
-    git:
-      url: https://github.com/linagora/flutter_typeahead.git
-      ref: twake-supported
```
Comment (hoangdat) [inline pubspec.yaml:50]: please remove the `#FIXME`

File: pubspec.yaml [hunk 5]
```diff
@@ -153,16 +150,20 @@ dependencies:
   media_kit_libs_video: ^1.0.1
   video_player: ^2.7.2
   js: ^0.6.7
-  super_clipboard: ^0.6.4
+  super_clipboard:
+    git:
+      url: https://github.com/superlistapp/super_native_extensions.git
+      ref: main
+      path: super_clipboard
```
Comment (hoangdat) [inline pubspec.yaml:157]: please add `FIXME` too

**Comment by hoangdat** [comment]
- [ ] no demo???

File: pubspec.yaml (hunk 5 above)
Comment (sherlockvn) [reply pubspec.yaml:157]: done

File: pubspec.yaml (hunk 4 above)
Comment (sherlockvn) [reply pubspec.yaml:50]: done

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**